cdap-common/src/test/java/co/cask/cdap/common/zookeeper/coordination/ResourceCoordinatorTest.java

```diff
@@ -236,4 +245,22 @@ public InetSocketAddress getSocketAddress() {
       }
     };
   }
+
+  private Collection<PartitionReplica> waitForChange(BlockingQueue<Collection<PartitionReplica>> assignmentQueue,
```

Comment:
It returns null because there is nothing in the queue even after 5 seconds. We don't need the retry logic, but simply having a longer timeout if it's just timing issue.

BlockingQueue.poll will return immediately if there is something in the queue.